test/test/CompilerTest.scala

```diff
@@ -89,7 +90,20 @@ abstract class CompilerTest {
     if (!generatePartestFiles || !partestableFile(prefix, fileName, extension, args ++ defaultOptions)) {
       if (runTest)
         log(s"WARNING: run tests can only be run by partest, JUnit just verifies compilation: $prefix$fileName$extension")
-      compileArgs((s"$filePath" :: args).toArray, expErrors)
+      if (args.contains("-rewrite")) {
```

Comment:
Instead of implicitly doing additional checks when one of the arg is `-rewrite`, I think it'd be better to ask for these checks explicitly, either have a `compileFileCheckRewrite` method or an additional argument `checkRewrite: Boolean = false` to `compileFile`.